### DISCUSSION TITLE
Hide the offcanvas on mobile back

### DISCUSSION THREAD

**Discussion by Jehong-Ahn** [twbs/bootstrap#36604] [Ideas]
Like other mobile apps.

**Comment by mdo** [comment]
Thoughts on this @GeoSot? Maybe this is more of an example in our docs vs specific functionality we need to build?

**Comment by GeoSot** [comment]
@Jehong-Ahn  can you be more explicit, to help me understand the idea?

(I suppose you suggest handling the swipe over offcanvas and close it on user gesture, but I am not sure if I have understood correct)

**Comment by Jehong-Ahn** [comment]
Sorry for bad english.

On my second thought this seems about android.

Android users expect the offcanvas get closed when he touch the device back button.

Some articles
- [4 Design Patterns That Violate “Back” Button UX Expectations – 59% of Sites Get It Wrong – Articles – Baymard Institute](https://baymard.com/blog/back-button-expectations)
- [Design with the web in mind. Back button, URL design, keyboard… | by Chloe Sanderson | UX Collective](https://uxdesign.cc/design-with-the-web-in-mind-d9f9df2e8812)

Some plugins
- [Back button add-on - mmenu.js documentation](https://mmenujs.com/docs/addons/back-button.html)

My demo

```
if ( navigator.userAgentData.mobile && navigator.userAgent.match(/Android/) ) {

  const $el = window['offcanvas-navbar'];

  $el.addEventListener('shown.bs.offcanvas', () => location.hash = '#offcanvas');
  
  $el.addEventListener('hidden.bs.offcanvas', () => location.hash==='#offcanvas' && history.back());
  
  window.onhashchange = event => {
    if ( event.oldURL.substr(-10) === '#offcanvas' ) bootstrap.Offcanvas.getInstance($el).hide();
  };

}

```

**Comment by Jehong-Ahn** [comment]
@GeoSot Read my comment pls.

**Comment by GeoSot** [comment]
Sorry for late response.
I have your message in my mind, and yes I agree with you, as it is the right user experience .

I 'll try to think how could be implemented it on 5.3

**Reply by Jehong-Ahn** [reply]
@GeoSot thanks 🤓

**Reply by GeoSot** [reply]
Keep in mind this PR https://github.com/twbs/bootstrap/pull/36648
